tests/test_pypi.py

```diff
@@ -809,13 +820,22 @@ def test_download_pkg_sdist(pkg_pytest):
 
 def test_ciso_recipe():
     recipe = GrayskullFactory.create_recipe(
-        "pypi", Configuration(name="ciso", version="0.1.0")
+        "pypi", Configuration(name="ciso", version="0.2.2")
     )
     assert sorted(recipe["requirements"]["host"]) == sorted(
-        ["cython", "numpy", "pip", "python"]
+        [
+            "cython >=3",
+            "numpy >=2.0.0rc1",
+            "oldest-supported-numpy",
+            "pip",
+            "python >=3.9",
+            "setuptools >=41.2",
+            "setuptools-scm",
+            "wheel",
+        ]
     )
     assert sorted(recipe["requirements"]["run"]) == sorted(
-        ["cython", "python", "<{ pin_compatible('numpy') }}"]
+        ["<{ pin_compatible('numpy') }}", "oldest-supported-numpy", "python >=3.9"]
```

Comment:
Also not sure why `oldest-supported-numpy` is added to `requirements/run`

Reply:
Hi @jakirkham , at first sight it seems odd, indeed, but it comes directly from the requirements in that package ``gsw``, it is not something that grayskull adds, 
https://github.com/TEOS-10/GSW-Python/blob/c3407368ef01d0b3d8e85f8cf7b94abae52bcd25/pyproject.toml#L6

that is a separated package on pypi
https://pypi.org/project/oldest-supported-numpy/

Reply:
Indeed this is actually for the [ciso](https://github.com/ioos/ciso) project that has `oldest-supported-numpy ` not only in build-system requirements , but in [dependencies]( https://github.com/ioos/ciso/blob/189ad38f84bc20176ad46b312fd312eddd548c53/pyproject.toml#L23C1-L26C2) as well:

```
dependencies = [
    "oldest-supported-numpy ; python_version < '3.9'",
    "numpy>=2.0.0rc1 ; python_version >= '3.9'",
]
```

Reply:
It's ok to have in `pyproject.toml`

However what this appears to be testing is whether this is added to `requirements/host` and `requirements/run`. The latter doesn't make sense

Reply:
it is going to appear in the requirements run because ``oldest-supported-numpy`` is specified in the dependencies section of the pyproject.toml. If it is was only in build section of pyproject.toml that would appear only in host then.

It is possible to do a workaround and always remove it from the requirements/run and always let it in host, but I am just not so sure if it covers all scenarios, what do you think?

Reply:
We should remove `oldest-supported-numpy` whenever it is seen. That's used by wheels specifically to configure builds (not sure why a wheel would use `oldest-supported-numpy` at runtime unless they are some kind of build tool)

In any event it doesn't make sense in the Conda case where we configure NumPy versions used in some `conda_build_config.yaml` or other [variants]( https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html )

On the Conda side, we need only add `numpy` (unconstrained) to `requirements/host` and copy over any `dependencies` value of `numpy` to `requirements/run`

Reply:
ok, gotcha
That is pretty much a new PR to do this workaround as it was not a requirement before.

Reply:
Gotcha thanks for the context! 🙏

The issue is `oldest-supported-numpy` doesn't exist in Conda. So if Grayskull adds it to recipes, they will be broken. Maintainers will need to know they can clean this out to fix the issue

In any event, we can discuss more in the new issue (thanks for raising! 🙏): https://github.com/conda/grayskull/issues/553